src/main/kotlin/ch/unibas/dmi/dbis/cottontail/model/values/complex/Complex.kt

```diff
@@ -0,0 +1,49 @@
+package ch.unibas.dmi.dbis.cottontail.model.values.complex
+
+import kotlin.math.sqrt
+
+/**
+ * Represents a complex number with single-precision 32-bit IEEE 754 floating point numbers.
+ *
+ * Inspired by:
+ *
+ *    - https://gist.github.com/simonvar/97ab9745e719c34772db4a10a8515d2c
+ *    - https://github.com/abdulfatir/jcomplexnumber/blob/master/com/abdulfatir/jcomplexnumber/ComplexNumber.java
+ *    - https://introcs.cs.princeton.edu/java/97data/Complex.java.html
+ *    - https://rosettacode.org/wiki/Arithmetic/Complex#Kotlin
+ *
+ * @author Manuel Huerbin
+ * @version 1.0
+ */
+class Complex(private val real: Float = 0.0f, private val imaginary: Float = 0.0f) {
```

Comment:
I would refrain from using classical object types as this creates a lot of overhead and remember, Cottontail DB processes a lot of these data types. Instead, I would use inline classes. They provide us with a "thin" wrapper around primitive values (that are still treated like primitive values by the JVM). 

There is several ways to approach this: 

- Either you implement the **Complex** class a an inline-class around a 2d float/double array (where first component is the real part and 2nd component is the imaginary part) 
- Or your directly implement ComplexValue and ComplexVectorValue using the suggested paradigm.